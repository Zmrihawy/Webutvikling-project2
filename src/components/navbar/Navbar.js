import React, {Component} from 'react'
import NavbarMedia from "./Navbarmedia";

class Navbar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            titles: ["Picture", "Sound", "Text"]
        }
    }

    render() {

        return (
            <div className="row">
                {this.state.titles.map((Item, index) => <NavbarMedia title={Item} key={index}/>)}
            </div>
        )
    }
}

export default Navbar;