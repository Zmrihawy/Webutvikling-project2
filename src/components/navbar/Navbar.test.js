import React from 'react';
import Navbar from './Navbar';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const state = {
            picture: {
                cubes: false,
                shapes: true,
                spirals: false
            }, 

            sounds: {
                romance: false,
                deep: false,
                wild: true
            },

            text: {
                boring: true,
                funny: false,
                random: false
            },

            others: {
                make_favourite: false,
                toggleFavourites: false,
                currentFavourites: [],
                chosenFavorite: null
            },

            combinations: {
                picture: [0, 1, 2, 3],
                sounds: [0, 1, 2, 3],
                text: [0, 1, 2, 3]
            }
        }


    const tree = renderer
    .create(
        <Navbar 
            setPicture={() => {}} 
            setText={() => {}} 
            setSound={() => {}}
            setFavourites={() => {}}
            globalState={state}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});