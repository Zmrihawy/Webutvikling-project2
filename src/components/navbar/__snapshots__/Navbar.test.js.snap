// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="_inner_navbar"
>
  <div
    className="container"
  >
    <h3>
      Picture
    </h3>
    <div
      className="checkbox"
    >
      <input
        checked={false}
        id="cubes"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="cubes"
      >
        cubes
      </label>
    </div>
    <div
      className="checkbox"
    >
      <input
        checked={true}
        id="shapes"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="shapes"
      >
        shapes
      </label>
    </div>
    <div
      className="checkbox"
    >
      <input
        checked={false}
        id="spirals"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="spirals"
      >
        spirals
      </label>
    </div>
  </div>
  <div
    className="container"
  >
    <h3>
      Sounds
    </h3>
    <div
      className="checkbox"
    >
      <input
        checked={false}
        id="romance"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="romance"
      >
        romance
      </label>
    </div>
    <div
      className="checkbox"
    >
      <input
        checked={false}
        id="deep"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="deep"
      >
        deep
      </label>
    </div>
    <div
      className="checkbox"
    >
      <input
        checked={true}
        id="wild"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="wild"
      >
        wild
      </label>
    </div>
  </div>
  <div
    className="container"
  >
    <h3>
      Text
    </h3>
    <div
      className="checkbox"
    >
      <input
        checked={true}
        id="boring"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="boring"
      >
        boring
      </label>
    </div>
    <div
      className="checkbox"
    >
      <input
        checked={false}
        id="funny"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="funny"
      >
        funny
      </label>
    </div>
    <div
      className="checkbox"
    >
      <input
        checked={false}
        id="random"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-container"
        htmlFor="random"
      >
        random
      </label>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="favorites-container"
    >
      <div
        className="dropdown"
      >
        <button
          className="dropbtn"
        >
          Favorites
          <i
            className="fas fa-caret-down"
          />
        </button>
        <div
          className="dropdown-content"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            None
          </a>
          <div>
            no favorites saved yet
          </div>
        </div>
      </div>
      <div
        className="make-favorite-container"
      >
        <form
          className="make-fav-input"
        >
          <label>
            <input
              onChange={[Function]}
              placeholder=" New Favourite Name ..."
              type="text"
              value=""
            />
          </label>
        </form>
        <button
          className="make-fav-btn"
          onClick={[Function]}
        >
          Save Favorite
        </button>
      </div>
    </div>
  </div>
</div>
`;
